third_party/move/move-vm/runtime/src/frame.rs

```diff
@@ -131,6 +132,21 @@ impl Frame {
         function_instantiation_handle_at
     );
 
+    pub(crate) fn get_or_compute_fingerprint(
+        &self,
+        idx: FunctionInstantiationIndex,
+        ty_args: &[Type],
+    ) -> TyArgsFingerprint {
+        let maybe_fingerprint = match self.function.owner() {
+            LoadedFunctionOwner::Script(script) => script.function_instantiation_fingerprint(idx.0),
+            LoadedFunctionOwner::Module(module) => module.function_instantiation_fingerprint(idx.0),
+        };
+        match maybe_fingerprint {
+            None => TyArgsFingerprint::from_ty_args(ty_args),
```

Comment:
This looks a bit weird, why would a fingerprint just based on ty_args be sufficient if those functions return None? Comments would help.